### PR TITLE
Attachment Feature

### PR DESCRIPTION
**Pull request by mantishK**
Added the ability to send files as attachments.

### CODE REVIEW

File: apostle/mail.py [hunk 1]
```diff
@@ -79,12 +83,18 @@ def to_recipient_dict(self):
 			'layout_id': self.layout_id,
 			'name': self.name,
 			'reply_to': self.reply_to,
-			'template_id': self.template_id
+			'template_id': self.template_id,
+            'attachments': self.attachments
```
Comment (snikch) [inline apostle/mail.py]: Indentation here is incorrect.

File: apostle/mail.py [hunk 2]
```diff
@@ -79,12 +83,18 @@ def to_recipient_dict(self):
...
+			if val == None or val == {} or val == []:
 				del recipient_dict[key]
 
 		return { self.email: recipient_dict }
 
+	def add_attachment(self, name, content):
+		self.attachments.append({
+			"name": name,
+			"data": base64.b64encode(content)
+		})
```
Comment (snikch) [inline apostle/mail.py]: Please ensure there's a trailing newline at the end of the files (for consistency).

**Review by snikch — CHANGES_REQUESTED** [review]
Thanks for this, looks good. Just a couple of style comments but the functionality looks great 👌

File: apostle/mail.py (hunk 2 above)
Comment (mantishK) [reply apostle/mail.py]: Fixed!

**Comment by mantishK** [comment]
@snikch Made the review changes

**Comment by snikch** [comment]
Great, thanks!

**Pull request merged**